pkg/hub/custommetrics/custommetrics.go

```diff
@@ -0,0 +1,198 @@
+	managedClusterMetric.WithLabelValues(managedCluster.GetName(), "cluster_id", "type", "version", managedCluster.GetLabels()["cloud"], hubID).Set(1)
+}
+
+func delCluster(obj interface{}) {
+	klog.Info("Deleting Managed Clusters ####################")
+
+	j, err := json.Marshal(obj.(*unstructured.Unstructured))
+	if err != nil {
+		klog.Warning("Error on ManagedCluster marshal.")
+	}
+	managedCluster := clusterv1.ManagedCluster{}
+	err = json.Unmarshal(j, &managedCluster)
+	if err != nil {
+		klog.Warning("Error on ManagedCluster unmarshal.")
+	}
+
+	klog.Infof("Managed Cluster name being removed: %s \n", managedCluster.GetName())
+
+	//TODO:
+	//get the actual values as mentioned here:
+	//https://github.com/open-cluster-management/perf-analysis/blob/master/Big%20Picture.md#acm-20-telemetry-data
+	managedClusterMetric.WithLabelValues(managedCluster.GetName(), "cluster_id", "type", "version", managedCluster.GetLabels()["cloud"], hubID).Set(0)
+
+}
+
+func fetchTestData(c dynamic.Interface) {
+
+	//TODO: Test - will be removed
+	managedClusterMetric.WithLabelValues("cluster_name", "cluster_id", "type", "version", "cluster_infrastructure_provider", "hub_id").Set(2.354)
+
+	klog.Infof("Getting data for Managed Clusters")
+
+	var stopper chan struct{}
+	informerRunning := false
+
+	dynamicFactory := dynamicinformer.NewDynamicSharedInformerFactory(c, 60*time.Second)
+	clusterInformer := dynamicFactory.ForResource(mcGVR).Informer()
+	clusterInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			addCluster(obj)
+		},
+		UpdateFunc: func(prev interface{}, next interface{}) {
+			klog.Info("Updating Managed Clusters ####################")
+		},
+		DeleteFunc: func(obj interface{}) {
+			delCluster(obj)
+		},
+	})
+
+	//Starting the informer
+	for {
+		if !informerRunning {
+			klog.Info("Starting cluster informer routine for cluster watch")
+			stopper = make(chan struct{})
+			informerRunning = true
+			go clusterInformer.Run(stopper)
+		}
+		//TODO: Check this setting
+		time.Sleep(60 * time.Second)
+	}
+
+}
+
+func MetricStart(port int32) {
+
+	var customMetricsPort = port
+	//registering the metrics to a custom registry
+	r := prometheus.NewRegistry()
+	r.MustRegister(managedClusterMetric)
+
+	dynClient, errClient := getDynClient()
+	if errClient != nil {
+		klog.Fatalf("Error received creating client %v \n", errClient)
+		panic(errClient.Error())
+	}
+
+	getStaticData(dynClient)
+
+	go fetchTestData(dynClient)
+
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(w, "Hello, %q, how are you ?\n", html.EscapeString(r.URL.Path))
+
+	})
+
+	http.Handle("/metrics", promhttp.HandlerFor(r, promhttp.HandlerOpts{}))
+
+	klog.Infof("Custom Metric Service started on port: " + fmt.Sprint(customMetricsPort))
+	klog.Fatal(http.ListenAndServe(":"+fmt.Sprint(customMetricsPort), nil))
```

Comment:
this looks inappropriate to me.  Just some testing code?  it's a little difficult to tell if this is a WIP or not.

You have a comment mentioning that 8443 is serving metrics. Is it via this or via the library-go server launch (which also holds healthz)

Reply:
this is gone as we are using the endpoint already created.